File: src/Plus/__to.php
<?php

namespace xltxlm\fileinfo\Plus;

use xltxlm\fileinfo\Features\Fileinfo_Read_Lines;
use xltxlm\fileinfo\Fileinfo;

/**
 * 在文件类之间随意转换的基础类;
 */
Trait __to
{
    use __to\__to_implements;


    public function __construct(string $Filepath = null)
    {
        if ($Filepath != null) {
            $this->setFilepath($Filepath);
        }
    }

    public function __Fileinfo_Read_Lines(): \xltxlm\fileinfo\Features\Fileinfo_Read_Lines
    {
        return new Fileinfo_Read_Lines($this->getFilepath());
    }

    public function __Fileinfo(): \xltxlm\fileinfo\Fileinfo
    {
        return new Fileinfo($this->getFilepath());
    }


}

File: go/Plus/__to.go
package Plus

type __to struct {
    /* 设置文件的路径 */
    Filepath string

}

func New__to(Filepath string) *__to{
    var this = new(__to)
    this.SetFilepath(Filepath);
    return this
}

func (this *__to)GetFilepath() string{
    return this.Filepath;
}

func (this *__to)SetFilepath(Filepath string) *__to{
    this.Filepath = Filepath;
    return this
}

/**
    */
func (this *__to)__Fileinfo_Read_Lines(){

}

/**
    */
func (this *__to)__Fileinfo(){

}

